python_modules/libraries/dagster-dbt/dagster_dbt/freshness_builder.py

```diff
@@ -0,0 +1,115 @@
+import datetime
+from typing import Sequence
+
+from dagster import (
+    AssetsDefinition,
+)
+from dagster._core.definitions.asset_check_factories.freshness_checks.last_update import (
+    build_last_update_freshness_checks,
+)
+from dagster._core.definitions.asset_check_factories.freshness_checks.time_partition import (
+    build_time_partition_freshness_checks,
+)
+from dagster._core.definitions.asset_check_factories.utils import DEFAULT_FRESHNESS_SEVERITY
+from dagster._core.definitions.asset_check_spec import AssetCheckSeverity
+from dagster._core.definitions.asset_checks import AssetChecksDefinition
+from dagster._core.definitions.time_window_partitions import TimeWindowPartitionsDefinition
+from dagster._core.errors import DagsterInvariantViolationError
+
+from dagster_dbt.utils import ASSET_RESOURCE_TYPES
+
+from .asset_utils import get_manifest_and_translator_from_dbt_assets
+
+
+def build_freshness_checks_from_dbt_assets(
+    dbt_assets: Sequence[AssetsDefinition],
+) -> Sequence[AssetChecksDefinition]:
+    """Returns a sequence of freshness checks constructed from the provided dbt assets.
+
+    Freshness checks can be configured on a per-model basis in the dbt project configuration.
+    For time-partitioned assets, the freshness check configuration mirrors that of the `build_time_partition_freshness_checks` function.
```

Comment:
Non-partitioned dbt assets are quite a bit more common than partitioned dbt assets, so we should lead with the former